opr-core/src/model/persistentoffermodel.ts

```diff
@@ -0,0 +1,923 @@
+          400
+        );
+      }
+      const t = await this.storage.createTransaction('READONLY');
+      const operations: Array<OfferPatch> = [];
+      // Check to see if there are any offers at the starting time. If there
+      // aren't any, start the diff with a 'clear' operation to make it clear
+      // we're sending the entire list of offers.
+      const hasOldOffers = await hasContents(
+        this.storage.getOffersAtTime(
+          t,
+          this.hostOrgUrl,
+          orgUrl,
+          payload.diffStartTimestampUTC!,
+          0,
+          1
+        )
+      );
+      if (!hasOldOffers) {
+        if (skipCount === 0) {
+          operations.push('clear');
+        } else {
+          skipCount--;
+        }
+      }
+      // Find all the change operations.
+      const offerIterable = this.storage.getChangedOffers(
+        t,
+        this.hostOrgUrl,
+        orgUrl,
+        payload.diffStartTimestampUTC!,
+        requestTimeUTC,
+        skipCount
+      );
+      const iterator: AsyncIterator<OfferVersionPair> =
+        offerIterable[Symbol.asyncIterator]();
+      let result;
+      while (
+        operations.length < pageSize &&
+        (result = await iterator.next()) &&
+        !result.done
+      ) {
+        const versionPair = result.value;
+        const jsonPatch = diffAsOfferPatch(
+          versionPair.newVersion,
+          versionPair.oldVersion
+        );
+        operations.push(jsonPatch);
+      }
+      let pageToken;
+      if (result && !result.done) {
+        pageToken = this.encodePageToken({
+          maxResultsPerPage: pageSize,
+          requestTimeUTC: requestTimeUTC,
+          skipCount: skipCount + operations.length,
+        } as DecodedListPageToken);
+      }
+      return {
+        responseFormat: 'DIFF',
+        resultsTimestampUTC: requestTimeUTC,
+        diff: operations,
+        nextPageToken: pageToken,
+      } as ListOffersResponse;
+    }
+  }
+
+  private getCorpusOfferLookup(
+    orgUrl: string,
+    transaction?: Transaction
+  ): OfferLookup {
+    return {
+      get: async (offerId: OfferId) => {
+        const sid = asStructuredId(offerId);
+        const t =
+          transaction ?? (await this.storage.createTransaction('READONLY'));
+        return await this.storage.getOfferFromCorpus(
+          t,
+          this.hostOrgUrl,
+          orgUrl,
+          sid.postingOrgUrl,
+          sid.id
+        );
+      },
+    } as OfferLookup;
+  }
+
+  async processUpdate(
+    fromOrgUrl: string,
+    update: OfferSetUpdate
+  ): Promise<void> {
+    const t = await this.storage.createTransaction();
+    const lookup = this.getCorpusOfferLookup(fromOrgUrl, t);
+    const now = this.clock.now();
+    if (update.delta) {
+      for await (const op of update.delta) {
+        const result = await applyOfferPatch(lookup, op);
+        switch (result.type) {
+          case 'CLEAR': {
+            // TODO: Clear the corpus
```

Comment:
Could this comment be removed? It looks like the corpus is being cleared in the following lines.

Reply:
Fixed